### ISSUE TITLE
[New Rule] AnySequence(dict.keys()) to AnySequence(dict)

### ISSUE THREAD

**Issue by janosh** [MartinThoma/flake8-simplify#181]
Would be nice if SIM118 could be expanded (or a new rule created) that not just covers iterating over dict keys but also converting them to any type of iterable:

```py
# bad
list(dict.keys())
tuple(dict.keys())
set(dict.keys())
[*dict.keys()]
{*dict.keys()}

# good
list(dict)
tuple(dict)
set(dict)
[*dict]
{*dict}
```

Related: https://github.com/charliermarsh/ruff/issues/4262
